StorageManager.persist() is not functional

szewai opened the pull request:
#### 338e1ced429a53ada0a2e52569a2d031a3a63361
<pre>
StorageManager.persist() is not functional
<a href="https://bugs.webkit.org/show_bug.cgi?id=256483">https://bugs.webkit.org/show_bug.cgi?id=256483</a>
rdar://109057460

Reviewed by Youenn Fablet.

StorageManager.persist() currently sets an in-memory flag of OriginStorageManager unconditionally and the flag does not
do anything. Now that we have implemented eviction based on quota, this patch ensures the eviction will check persited
flag and skip deleting the origin if its flag is set. Also, this patch ensures persisted flag is stored to disk by
creating a persisted file.

* LayoutTests/imported/w3c/web-platform-tests/storage/storagemanager-persist-persisted-match.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/storage/storagemanager-persist-persisted-match.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/storage/storagemanager-persist-persisted-match.https.any.js: Added.
(promise_test.async t):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::persistedFilePath):
(WebKit::NetworkStorageManager::donePrepareForEviction):
(WebKit::NetworkStorageManager::performEviction):
(WebKit::NetworkStorageManager::persistedInternal):
(WebKit::NetworkStorageManager::persisted):
(WebKit::NetworkStorageManager::persist):
(WebKit::NetworkStorageManager::resetStoragePersistedState):
(WebKit::NetworkStorageManager::deleteDataOnDisk):
(WebKit::NetworkStorageManager::evictDataByTopOrigin): Deleted.
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.h:
* Source/WebKit/NetworkProcess/storage/OriginStorageManager.cpp:
(WebKit::OriginStorageManager::setPersisted): Deleted.
* Source/WebKit/NetworkProcess/storage/OriginStorageManager.h:
(WebKit::OriginStorageManager::persisted const): Deleted.

Canonical link: <a href="https://commits.webkit.org/263930@main">https://commits.webkit.org/263930@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a70d87048f4563e9b764ee53bb790a35392776d3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6128 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6313 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6496 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7691 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6474 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6532 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6266 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9348 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6238 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6253 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5540 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7753 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3728 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5521 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/13425 "4 flakes 174 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5593 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5599 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/7832 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6073 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4970 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5493 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/5450 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1456 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9636 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5861 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on LayoutTests/imported/w3c/web-platform-tests/storage/storagemanager-persist-persisted-match.https.any.js:
```diff
@@ -0,0 +1,13 @@
+
+promise_test(function() {
+    var persistResult = false;
+    return navigator.storage.persist().then(function(result) {
```
async/await would be more natural.
Since it is a WPT test, we should upstream these changes before merging this PR or move it to LayoutTests/http/wpt/.

szewai committed:
StorageManager.persist() is not functional
https://bugs.webkit.org/show_bug.cgi?id=256483
rdar://109057460

Reviewed by Youenn Fablet.

StorageManager.persist() currently sets an in-memory flag of OriginStorageManager unconditionally and the flag does not
do anything. Now that we have implemented eviction based on quota, this patch ensures the eviction will check persited
flag and skip deleting the origin if its flag is set. Also, this patch ensures persisted flag is stored to disk by
creating a persisted file.

* LayoutTests/imported/w3c/web-platform-tests/storage/storagemanager-persist-persisted-match.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/storage/storagemanager-persist-persisted-match.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/storage/storagemanager-persist-persisted-match.https.any.js: Added.
(promise_test.async t):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::persistedFilePath):
(WebKit::NetworkStorageManager::donePrepareForEviction):
(WebKit::NetworkStorageManager::performEviction):
(WebKit::NetworkStorageManager::persistedInternal):
(WebKit::NetworkStorageManager::persisted):
(WebKit::NetworkStorageManager::persist):
(WebKit::NetworkStorageManager::resetStoragePersistedState):
(WebKit::NetworkStorageManager::deleteDataOnDisk):
(WebKit::NetworkStorageManager::evictDataByTopOrigin): Deleted.
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.h:
* Source/WebKit/NetworkProcess/storage/OriginStorageManager.cpp:
(WebKit::OriginStorageManager::setPersisted): Deleted.
* Source/WebKit/NetworkProcess/storage/OriginStorageManager.h:
(WebKit::OriginStorageManager::persisted const): Deleted.

Canonical link: https://commits.webkit.org/263930@main
